dot/core/service.go

```diff
@@ -55,8 +52,10 @@ type Service struct {
 	codeHash common.Hash
 
 	// Current BABE session
-	bs              *babe.Session
-	isBabeAuthority bool
+	bs          *babe.Session
+	isAuthority bool
+	epochNumber uint64   // epoch number of current epoch
+	firstBlock  *big.Int // block number of first block in current epoch
```

Comment:
added two new params to track epoch and first block in core service

Reply:
I'm wondering if we should have a map of epoch number -> first block, how would this work when syncing lots of epochs? when syncing, babe sessions don't get regenerated every epoch, it just waits until syncing is done, so I'm concerned about this being incorrect for syncing

Reply:
Can you create a separate issue for this?